File: Skip_Backend/src/main/java/com/simplife/skip/models/Viaje.java
package com.simplife.skip.models;

import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import javax.persistence.*;
import java.io.Serializable;
import java.time.LocalDate;
import java.time.LocalTime;

@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
@Table(name = "viaje")
@Entity
public class Viaje implements Serializable {

    private static final long serialVersionUID = 1L;

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "viaje_id")
    private Long id;

    @Column(name = "conductor_id")
    private Usuario conductor;

    @Column(name = "descripcion")
    private String descripcion;

    @Column(name = "fecha_publicacion")
    private LocalDate fechaPublicacion;

    @Column(name = "hora_inicio")
    private LocalTime horaInicio;

    @Column(name = "hora_llegada")
    private LocalTime horaLlegada;

    @Column(name = "hora_publicacion")
    private LocalTime horaPublicacion;

    @Column(name = "visualizacion_habilitada")
    private boolean visualizacionHabilitada;

    @Column(name = "ruta_id")
    private Ruta ruta;

    //Iniciado - Finalizado
    @Column(name = "estado_viaje")
    private String estadoViaje;



}

File: Skip_Android/app/src/main/java/com/simplife/skip/models/Viaje.kt
package com.simplife.skip.models

import java.io.Serializable

data class Viaje (
    var id: Long,
    var conductor: Usuario,
    var ruta: Ruta,
    var descripcion: String,
    var fechaPublicacion: String,
    var fechaViaje: String,
    var horaInicio: String,
    var horaLlegada: String,
    var visualizacionHabilitada: Boolean,
    var numeroPasajeros: Int,
    var estadoViaje: String,
    var horaPublicacion: String,
    var estadoTabla: Boolean

) : Serializable {

    override fun toString(): String {
        return "BlogPost(title='$fechaViaje', image='$fechaPublicacion', username='$horaPublicacion')"
    }


}

